lms/djangoapps/grades/api/views.py

```diff
@@ -143,13 +160,26 @@ def get(self, request, course_id):
                 error_code='user_mismatch'
             )
 
+        # Returns a 404 response if course_id is invalid, or request.user is not enrolled in the course
         course = self._get_course(course_id, request.user, 'load')
         if isinstance(course, Response):
+            # A student can only access her own grade status info
+            if request.user.username != username:
+                return user_mismatch_error()
```

Comment:
Ah sorry, I think I was trying to catch a case that didn't exist.  https://github.com/edx/edx-platform/pull/14710/commits/eb11c98ce6d52cb84b3f02b79a0b3e71b9224a63#diff-dd3ce32b0bf0e8e67e0d10618ea0d4f7R147 removes this check.